components/suggest/src/query.rs

```diff
@@ -118,4 +120,122 @@ impl SuggestionQuery {
             ..self
         }
     }
+
+    // Other Functionality
+
+    /// Parse the `keyword` field into a set of keywords.
+    ///
+    /// This is used when passing the keywords into an FTS search.  It:
+    ///   - Strips out any `():^*"` chars.  These are typically used for advanced searches, which
+    ///     we don't support and it would be weird to only support for FTS searches, which
+    ///     currently means Fakespot searches.
+    ///   - Splits on whitespace to get a list of individual keywords
+    ///
+    #[cfg(feature = "fakespot")]
+    pub(crate) fn parse_keywords(&self) -> Vec<&str> {
+        self.keyword
+            .split(|c: char| {
+                c.is_whitespace()
+                    || c == '('
+                    || c == ')'
+                    || c == ':'
+                    || c == '^'
+                    || c == '*'
+                    || c == '"'
+            })
+            .filter(|s| !s.is_empty())
+            .collect()
+    }
+
+    /// Create an FTS query term for our keyword(s)
+    #[cfg(feature = "fakespot")]
+    pub(crate) fn fts_query(&self) -> String {
+        // Create a string to hold the formatted query, try to make the capacity large enough so
+        // that we don't need re-allocations
+        let mut fts_query = String::with_capacity(self.keyword.len() * 2);
+        // Quote and append each term from `query`
+        let keywords = self.parse_keywords();
+
+        if keywords.is_empty() {
+            return String::from("\"\"");
+        }
+        for keyword in keywords[..keywords.len() - 1].iter() {
+            fts_query.push('"');
```

Comment:
Per [doc](https://www.sqlite.org/fts5.html#fts5_strings), an FTS string can be enclosed by double quotes as a whole, e.g. `match '"foo bar baz"'`, no need to wrap each token by double quotes, e.g `match '"foo" "bar" "baz"'`, correct?

Reply:
Those docs are very confusing to me, but I think that makes the entire string a phrase, which needs to be matched together, without any tokens in the middle.  If I change the code to quote things differently, then [this test](https://github.com/mozilla/application-services/blob/a3c093f7afc72df86116097a649f0195bdf300e7/components/suggest/src/store.rs#L2114-L2117) fails.

Reply:
I'm still not totally sure what's going on, but the code appears to work when I quote each term separately so I'm going to leave it like that.   I'm happy to continue to iterate on this one, I definitely think there's room for improvement here.